sky/packages/sky/lib/src/material/material_app.dart

```diff
@@ -73,18 +75,24 @@ class _MaterialAppState extends State<MaterialApp> {
   }
 
   Widget build(BuildContext context) {
-    return new Theme(
-      data: config.theme ?? new ThemeData.fallback(),
-      child: new DefaultTextStyle(
-        style: _errorTextStyle,
-        child: new DefaultAssetBundle(
-          bundle: _defaultBundle,
-          child: new Title(
-            title: config.title,
-            child: new Navigator(
-              key: _navigator,
-              routes: config.routes,
-              onGenerateRoute: config.onGenerateRoute
+    return new SizeObserver(
+      onSizeChanged: (Size size) => setState(() { _size = size; }),
```

Comment:
The "size of the app" isn't necessarily a well-defined concept.  The "app"
could be a full-screen activity or a widget in some larger activity.  I
think that policy over what layout to choose (eg. based on breakpoints and
other information) should be determined at a higher level and propagated
down through the app.  Maybe at the level of the router.

Jeff.

On Tue, Oct 27, 2015 at 2:05 PM Ian Hickson <notifications@github.com>
wrote:

> In sky/packages/sky/lib/src/material/material_app.dart
> <https://github.com/flutter/engine/pull/1831#discussion_r43186992>:
>
> > @@ -73,18 +75,24 @@ class _MaterialAppState extends State<MaterialApp> {
> >    }
> >
> >    Widget build(BuildContext context) {
> > -    return new Theme(
> > -      data: config.theme ?? new ThemeData.fallback(),
> > -      child: new DefaultTextStyle(
> > -        style: _errorTextStyle,
> > -        child: new DefaultAssetBundle(
> > -          bundle: _defaultBundle,
> > -          child: new Title(
> > -            title: config.title,
> > -            child: new Navigator(
> > -              key: _navigator,
> > -              routes: config.routes,
> > -              onGenerateRoute: config.onGenerateRoute
> > +    return new SizeObserver(
> > +      onSizeChanged: (Size size) => setState(() { _size = size; }),
>
> the size of the whole app, or the size of subsegments?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/flutter/engine/pull/1831/files#r43186992>.
>